path_check.go

```diff
@@ -0,0 +1,53 @@
+package kubesecrets
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	checkPath            = "check"
+	checkHelpSynopsis    = `Checks the Kubernetes configuration is valid.`
+	checkHelpDescription = `Checks the Kubernetes configuration is valid, checking if required environment variables are set.`
+)
+
+var envVarsToCheck = []string{"KUBERNETES_SERVICE_HOST", "KUBERNETES_SERVICE_PORT_HTTPS"}
+
+func (b *backend) pathCheck() *framework.Path {
+	return &framework.Path{
+		Pattern: checkPath + "/?$",
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathCheckRead,
+			},
+		},
+		HelpSynopsis:    checkHelpSynopsis,
+		HelpDescription: checkHelpDescription,
+	}
+}
+
+func (b *backend) pathCheckRead(_ context.Context, _ *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	var missing []string
+	for _, key := range envVarsToCheck {
+		val := os.Getenv(key)
+		if val == "" {
+			missing = append(missing, key)
+		}
+	}
```

Comment:
There is also the `getK8sURLFromEnv()` function that would essentially provide the same information, though it doesn't currently return which variable(s) is unset if it fails.

https://github.com/hashicorp/vault-plugin-secrets-kubernetes/blob/eb48a93eda31a500d846ed2af7737d05d5e755dd/path_config.go#L232-L239

Reply:
Cool. I do slightly prefer to say which one failed I think?

Reply:
Sure, up to you, just thinking out loud 👍